container/image.bzl

```diff
@@ -124,84 +124,168 @@ def _image_config(
         else:
             labels[label] = fname
 
-    args = [
-        "--output=%s" % config.path,
-    ] + [
-        "--manifestoutput=%s" % manifest.path,
-    ] + [
-        "--entrypoint=%s" % x
-        for x in entrypoint
-    ] + [
-        "--command=%s" % x
-        for x in cmd
-    ] + [
-        "--ports=%s" % x
-        for x in ctx.attr.ports
-    ] + [
-        "--volumes=%s" % x
-        for x in ctx.attr.volumes
-    ] + [
-        "--null_entrypoint=%s" % null_entrypoint,
-    ] + [
-        "--null_cmd=%s" % null_cmd,
-    ]
-    if creation_time:
-        args += ["--creation_time=%s" % creation_time]
-    elif ctx.attr.stamp:
-        # If stamping is enabled, and the creation_time is not manually defined,
-        # default to '{BUILD_TIMESTAMP}'.
-        args += ["--creation_time={BUILD_TIMESTAMP}"]
-
-    args += ["--labels=%s" % "=".join([key, value]) for key, value in labels.items()]
-    args += ["--env=%s" % "=".join([
-        ctx.expand_make_variables("env", key, {}),
-        ctx.expand_make_variables("env", value, {}),
-    ]) for key, value in env.items()]
-
-    if ctx.attr.user:
-        args += ["--user=" + ctx.attr.user]
-    if workdir:
-        args += ["--workdir=" + workdir]
-
-    inputs = layer_names
-    for layer_name in layer_names:
-        args += ["--layer=@" + layer_name.path]
-
-    if ctx.attr.label_files:
-        inputs += ctx.files.label_files
-
-    if base_config:
-        args += ["--base=%s" % base_config.path]
-        inputs += [base_config]
-
-    if base_manifest:
-        args += ["--basemanifest=%s" % base_manifest.path]
-        inputs += [base_manifest]
-
-    if operating_system:
-        args += ["--operating_system=%s" % operating_system]
-
-    if ctx.attr.stamp:
-        stamp_inputs = [ctx.info_file, ctx.version_file]
-        args += ["--stamp-info-file=%s" % f.path for f in stamp_inputs]
-        inputs += stamp_inputs
-
-    if ctx.attr.launcher_args and not ctx.attr.launcher:
-        fail("launcher_args does nothing when launcher is not specified.", attr = "launcher_args")
-    if ctx.attr.launcher:
-        args += [
-            "--entrypoint_prefix=%s" % x
-            for x in ["/" + ctx.file.launcher.basename] + ctx.attr.launcher_args
+    if ctx.attr.legacy_create_image_config:
```

Comment:
Suggest creating two functions that take "args" & "inputs" and adds the relevant arguments & inputs for the python & go cases. This way, the core impl function will be smaller. So something like:
```
args = []
inputs = []
exec = None
if ctx.attr.legacy_create_image_config:
  _add_legacy_args(args, inputs, manifest, entrypoint, ...)
  exec = ctx.executable.create_image_config
else:
  _add_go_args(args, inputs, manifest, entrypoint, ...)
  exec = ctx.executable.go_create_image_config
ctx.actions.run(
        executable = exec,
        arguments = args,
        inputs = inputs,
        outputs = [config, manifest],
        use_default_shell_env = True,
        mnemonic = "ImageConfig",
    )
```